### PR TITLE
feat: add upload files mutations

### PR DESCRIPTION
**Pull request by pyphilia**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,8 +32,8 @@
   "devDependencies": {
     "@commitlint/cli": "19.3.0",
     "@commitlint/config-conventional": "19.2.2",
-    "@graasp/sdk": "4.13.0",
-    "@graasp/translations": "1.28.1",
+    "@graasp/sdk": "4.14.0",
+    "@graasp/translations": "github:graasp/graasp-translations#upload-files",
```
Comment (spaenleh) [inline package.json:36]: Do not forget to update with fixed version

File: src/item/create/mutations.test.ts [hunk 2]
```diff
@@ -0,0 +1,487 @@
...
+      await waitForMutation();
+    });
+
+    expect(
+      queryClient.getQueryState(itemKeys.allAccessible())?.isInvalidated,
+    ).toBeTruthy();
+  });
+});
+
+describe('useUploadFilesFeedbackFeedback', () => {
```
Comment (spaenleh) [inline src/item/create/mutations.test.ts:240]:
```suggestion
describe('useUploadFilesFeedback', () => {
```

File: src/item/create/mutations.ts [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+  );
+};
+
+/**
+ * @deprecated use useUploadFiles
+ * this mutation is used for its callback and invalidate the keys
+ * @param {UUID} id parent item id where the file is uploaded in
+ * @param {error} [error] error ocurred during the file uploading
+ */
+export const useUploadFilesFeedback =
```
Comment (spaenleh) [inline src/item/create/mutations.ts:63]: Will we be able to remove this one ?

File: src/item/routes.ts [hunk 4]
```diff
@@ -23,19 +23,30 @@ export const buildGetAccessibleItems = (
...
+export const buildPostItemRoute = (parentId?: UUID, previousItemId?: UUID) => {
+  const url = ITEMS_ROUTE;
+  const params = new URLSearchParams();
   if (parentId) {
-    url += `?parentId=${parentId}`;
+    params.set('parentId', parentId);
   }
-  return url;
+  if (previousItemId) {
+    params.set('previousItemId', previousItemId);
```
Comment (spaenleh) [inline src/item/routes.ts:33]: I am not sure to understand what this `previousItemId` is ?

File: src/item/routines.ts [hunk 5]
```diff
@@ -10,7 +10,7 @@ export const copyItemRoutine = createRoutine('COPY_ITEM');
 export const copyItemsRoutine = createRoutine('COPY_ITEMS');
 export const editItemRoutine = createRoutine('EDIT_ITEM');
 export const deleteItemsRoutine = createRoutine('DELETE_ITEMS');
-export const uploadFileRoutine = createRoutine('UPLOAD_FILE');
+export const uploadFilesRoutine = createRoutine('UPLOAD_FILE');
```
Comment (spaenleh) [inline src/item/routines.ts:13]:
```suggestion
export const uploadFilesRoutine = createRoutine('UPLOAD_FILES');
```

File: src/item/thumbnail/api.ts [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+import { verifyAuthentication } from '../../api/axios.js';
+import { DEFAULT_THUMBNAIL_SIZE } from '../../config/constants.js';
+import { PartialQueryConfigForApi } from '../../types.js';
+import {
+  buildDeleteItemThumbnailRoute,
+  buildDownloadItemThumbnailRoute,
+  buildUploadItemThumbnailRoute,
+} from '../routes.js';
+
+export const downloadItemThumbnail = async (
```
Comment (spaenleh) [inline src/item/thumbnail/api.ts:14]: Is this still used ?

File: src/item/thumbnail/mutations.test.ts [hunk 7]
```diff
@@ -0,0 +1,302 @@
...
+
+    expect(mockedNotifier).toHaveBeenCalledWith(
+      expect.objectContaining({
+        type: deleteItemThumbnailRoutine.SUCCESS,
+      }),
+    );
+  });
+});
+
+describe('useUploadFiles', () => {
```
Comment (spaenleh) [inline src/item/thumbnail/mutations.test.ts:180]: Update test name ?

**Review by spaenleh — APPROVED** [review]
Looks good, just a few comments.

File: src/item/create/mutations.ts (hunk 3 above)
Comment (pyphilia) [reply src/item/create/mutations.ts:63]: I think it's quite safe to remove it, I've set it to deprecated and we can remove at some point.

**Pull request merged**